langchain/chains/base.py

```diff
@@ -119,16 +119,25 @@ def apply(self, input_list: List[Dict[str, Any]]) -> List[Dict[str, str]]:
         """Call the chain on all inputs in the list."""
         return [self(inputs) for inputs in input_list]
 
-    def run(self, text: str) -> str:
-        """Run text in, text out (if applicable)."""
-        if len(self.input_keys) != 1:
-            raise ValueError(
-                f"`run` not supported when there is not exactly "
-                f"one input key, got {self.input_keys}."
-            )
+    def run(self, *args: str, **kwargs: str) -> str:
+        """Run the chain as text in, text out or multiple variables, text out."""
         if len(self.output_keys) != 1:
             raise ValueError(
                 f"`run` not supported when there is not exactly "
-                f"one output key, got {self.output_keys}."
+                f"one output key. Got {self.output_keys}."
             )
-        return self({self.input_keys[0]: text})[self.output_keys[0]]
+
+        if args and not kwargs:
+            if len(args) != len(self.input_keys):
```

Comment:
thoughts on validating that len == 1 here? otherwise this relies on the order of the inputs keys which seems a bit fragile

Reply:
Yeah, I went back and forth on this but ultimately decided it's fine to use either kwargs or args for multiple input keys. If not using kwargs, user has to pay attn to ordering of input keys. People have to do the same thing for function args too. Happy to restrict to one arg if you feel strongly

Reply:
the difference in my mind is that function args are pretty easily inspectable/obvious (eg you can shift tab, command p, etc) and that if you mess up the order its possible/likely that some other error would occur (due to type differences, etc)

here the order is not obvious, nad there would  likely never be a loud error (since boht type string) so it would silently mess it up... i feel medium strong about this

Reply:
Fair enough. Updated PR